Functionalization: Only include headers for required ops

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69840
* #68462
* #68693
* #68692
* #68691
* #68697
* __->__ #68690
* #68689
* #68714
* #68688
* #68687
* #68247

RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

peterbell10 committed:
Functionalization: Only include headers for required ops

RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

[ghstack-poisoned]

albanD reviewed (approved):
SGTM

FYI @bdhirsh

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596275).

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596275).

peterbell10 committed:
Rebase to fix merge conflicts on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596275).

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Fix mypy issues on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Commit to re-run CI on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Functionalization: Only include headers for required ops"


RegisterFunctionalization.cpp is a shared file, so only including the
required operators means a single operator change only requires 1
shard to be rebuilt instead of all of them.

Differential Revision: [D32596275](https://our.internmc.facebook.com/intern/diff/D32596275)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596275).
